addons/web/static/src/scss/bs_mixins_overrides.scss

```diff
@@ -51,6 +51,12 @@ $o-color-extras-nesting-selector: '&' !default;
         $-yiq-threshold-met: alpha($color) > $yiq-min-opacity-threshold;
 
         $-yiq-color: if($text-color, $text-color, if($-yiq-threshold-met, color-contrast($color, $main-background: $background), null));
+
+        // Generate utility classes for bg-* classes to then use them within
+        // the .badge design definition (see website.scss)
+        --bg-light: #{$color};
+        --color-light: #{$-yiq-color};
```

Comment:
- Indeed, the comment is misleading, we are generating CSS variables, I'll adapt
- This is indeed not optimal, but it'll be adapted within the Master fw-port and we'll use --background-color there ([see here](https://github.com/odoo/odoo/pull/188670#issuecomment-2527216429))
- I do think it's fine to keep the old design for these scenario. The current badge design requires a text-bg-X class to provide the new design. Using bg-X will return the old one. Since text-bg-X doesn't exists for those colors, I do think it's fine. We had to keep the old design for some other components as well such as badges within dropdown, product cart, ...